examples/plotting/grid_plotting.py

```diff
@@ -0,0 +1,38 @@
+"""
+==========================
+Plotting a coordinate grid
+==========================
+
+This example demonstrates how you can draw a grid of solar coordinates on top
+of a map.
+"""
+import matplotlib.pyplot as plt
+
+import sunpy.data.sample
+import sunpy.map
+
+################################################################################
+# Let's start with a sample AIA image.
+aia_map = sunpy.map.Map(sunpy.data.sample.AIA_171_IMAGE)
+
+################################################################################
+# Now we can plot the image, and add grids in heliographic Stonyhurst and
+# heliographic Carrington coordinates.
+fig = plt.figure()
+ax1 = fig.add_subplot(121, projection=aia_map)
+aia_map.plot(axes=ax1)
+aia_map.draw_grid(axes=ax1, system='stonyhurst')
+
+
+ax2 = fig.add_subplot(122, projection=aia_map)
+aia_map.plot(axes=ax2)
+aia_map.draw_grid(axes=ax2, system='carrington')
+
+for ax in [ax1, ax2]:
+    ax.set_title('')
+    ax.set_xlim(0, 400)
+    ax.set_ylim(0, 400)
+
+
+fig.subplots_adjust(wspace=0.5)
```

Comment:
Do some more adjusting of the plotting (or just specify a wider figure size) so that the labels don't run all over each other:
![sphx_glr_grid_plotting_001](https://user-images.githubusercontent.com/991759/158727823-bddd7cd3-0b65-4a41-9add-447b5da29ce7.png)

Reply:
How about now?
![sphx_glr_grid_plotting_001](https://user-images.githubusercontent.com/6197628/159177902-39bf52d9-c3f6-4147-8789-edd77e5e8413.png)